packages/block-library/src/table/editor.scss

```diff
@@ -63,6 +63,10 @@
 		}
 	}
 
+	.components-base-control__field {
+		margin-bottom: 0;
```

Comment:
This removes the bottom margins of the input fields, allowing the Inputs + Button to align correctly.